common/src/main/java/whocraft/tardis_refined/common/network/messages/ChangeShellMessage.java

```diff
@@ -58,8 +58,9 @@ public void handle(MessageContext context) {
         Optional<ServerLevel> level = Optional.ofNullable(context.getPlayer().getServer().levels.get(resourceKey));
         level.flatMap(TardisLevelOperator::get).ifPresent(y -> {
             if(Upgrades.CHAMELEON_CIRCUIT_SYSTEM.get().isUnlocked(y.getUpgradeHandler()) && y.getExteriorManager().hasEnoughFuelForShellChange()) {
-                y.setShellTheme(this.shellTheme, false);
                 y.getAestheticHandler().setShellPattern(pattern);
+                y.setShellTheme(this.shellTheme, false);
```

Comment:
Wouldn't the pattern get reset if it's not compatible with the shell theme?

Reply:
So the way we had it before was:

- Set Shell theme and do logic for updating
- Set Pattern

Which resulted in the pattern messing up a lot

Now we do:
- Set Pattern
- Set Shell theme and logic


The pattern setter was literally just that, with not updating or anything done for it to the client or blockentity - the fact it accidentally semi-worked this whole time baffles me